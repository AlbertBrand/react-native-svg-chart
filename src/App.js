// @flow
import React, { Component } from 'react';
import {
  StyleSheet,
  View,
} from 'react-native';
import LineChartContainer from './LineChartContainer';

const style = StyleSheet.create({
  chart: {
    paddingVertical: 100,
    width: 320,
    height: 300,
  },
});

export default class App extends Component {
  render() {
    const chartProps = {
      data: [
        [0, 10],
        [1, 20],
        [2, 40],
        [3, 23],
        [4, 67],
        [5, 1],
        [6, 54],
        [7, 96],
        [8, 23],
        [9, 44],
      ],
      options: {
        dashLineColor: '#d0d0d0',
        dataLineColor: '#0000ff',
        textStyle: {
          color: '#000000',
          fontFamily: 'arial',
          fontSize: 12,
        },
        xAccessor: val => val[0],
        yAccessor: val => val[1],
        xLabel: x => String(x),
        yLabel: y => String(y),
      },
    };
    return (
      <View>
        <LineChartContainer {...chartProps} style={style.chart} />
      </View>
    );
  }
}

