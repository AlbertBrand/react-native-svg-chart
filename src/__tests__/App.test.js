// @flow
// eslint-env jest
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

jest.mock('../LineChart', () => 'LineChart');

it('renders correctly', () => {
  const tree = renderer.create(
    <App />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
