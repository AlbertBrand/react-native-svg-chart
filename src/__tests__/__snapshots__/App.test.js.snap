exports[`test renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "height": 300,
    }
  } />
`;
